### PR TITLE
Allow sweeping cache without cache_size and with cache_size_bytes only.

### PR DESCRIPTION
**Pull request by yuseitahara**
Hello. This PR is for zopefoundation/ZODB#231 . cache_size_bytes works with cache_size=0,  no need to set cache_size to a large value.

### CODE REVIEW

**Comment by yuseitahara** [comment]
I will fix failing tests.

**Comment by yuseitahara** [comment]
All checks have passed. Is it ok to merge?

File: persistent/tests/test_picklecache.py [hunk 1]
```diff
@@ -1070,6 +1070,50 @@ def test_new_ghost_obj_already_in_cache(self):
...
+                                               kind=MyPersistent,
+                                               state=UPTODATE)
+            # must start 0, ZODB sets it AFTER updating the size
+            o._Persistent__size = 0
+            cache.update_object_size_estimation(oid, 1)
+            o._Persistent__size = 1
+            del o # leave it only in the cache
+
+        self.assertEqual(cache.cache_non_ghost_count, 100)
+        self.assertEqual(cache.total_estimated_size, 64*100)
```
Comment (jamadden) [inline persistent/tests/test_picklecache.py]: One space around operators, please, in all occurrences (e.g., `64 * 100`).

File: persistent/picklecache.py [hunk 2]
```diff
@@ -336,7 +336,10 @@ def _sweep(self, target, target_size_bytes=0):
         i = -1
         to_eject = []
         for value in self.ring:
-            if self.non_ghost_count <= target and (self.total_estimated_size <= target_size_bytes or not target_size_bytes):
+            if ((target or target_size_bytes) and
+               (not target or self.non_ghost_count <= target) and
```
Comment (jamadden) [inline persistent/picklecache.py]: Line breaks *before* the binary operators, please. (Per this [pep-8 update](https://github.com/python/peps/commit/7690a00d478866eb3cae3d82bbe1e80199a7392f)) (That is, `and` and `or` belong at the beginning of the line.) E.g.,
```python
if (thing 
     and other_thing
     or this_thing):
```

**Comment by yuseitahara** [comment]
Thank you for reviewing. I fixed them.

**Comment by icemac** [comment]
@yuseitahara An additional change log entry would be nice. After that feel free to merge your pull request.

**Pull request merged**